packages/rum-core/src/domain/view/viewMetrics/trackInteractionToNextPaint.ts

```diff
@@ -46,7 +47,14 @@ export function trackInteractionToNextPaint(
     }
   }
 
-  const { getViewInteractionCount } = trackViewInteractionCount(viewLoadingType)
+  const { getViewInteractionCount, stopViewInteractionCount } = trackViewInteractionCount(viewLoadingType)
+
+  let viewEnd = Infinity as RelativeTime
+  lifeCycle.subscribe(LifeCycleEventType.VIEW_ENDED, (data: ViewEndedEvent) => {
+    viewEnd = data.endClocks.relative
+    stopViewInteractionCount()
+  })
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏using the `VIEW_ENDED` event here is a bit confusing, because in some case `trackInteractionToNextPaint` is "driven" by the caller, and in some case it "drives" itself. Maybe a `setViewEnd(RelativeTime)` (to mimic the `setLoadEvent` logic) would be nicer, and we could remove the `LifeCycle` dependency when tackling https://datadoghq.atlassian.net/browse/RUM-958